packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/generate-auth-trigger-template.ts

```diff
@@ -122,11 +122,13 @@ const createCustomResource = (stack: cdk.Stack, authTriggerConnections: AuthTrig
   // The custom resource that uses the provider to supply value
   // Passing in a nonce parameter to ensure that the custom resource is triggered on every deployment
   // eslint-disable-next-line no-new
-  new CustomResource(stack, 'CustomAuthTriggerResource', {
+  const customResource = new CustomResource(stack, 'CustomAuthTriggerResource', {
     serviceToken: authTriggerFn.functionArn,
     properties: { userpoolId: userpoolId.valueAsString, lambdaConfig: authTriggerConnections, nonce: uuid() },
     resourceType: 'Custom::CustomAuthTriggerResourceOutputs',
   });
+
+  customResource.node.addDependency(authTriggerFn);
```

Comment:
should we also add the policy from line 114 as a dependency, just in case?
It looks like the custom resource might already be waiting on the authTriggerFn, but not the policy
![image](https://user-images.githubusercontent.com/110861985/204946006-a404ada4-cad3-4a96-86b1-98a5d28cc0d3.png)

Reply:
That dependsOn will add all the dependencies of the dependency added... I'll add a test to make it clearer. (As discussed off GitHub)